app/main/lib/shared_models/shared_model.py

```diff
@@ -12,141 +12,159 @@
 
 from flask import current_app as app
 
-Task = namedtuple('Task', 'task_id task_type task_package')
+Task = namedtuple("Task", "task_id task_type task_package")
+
 
 class SharedModel(object):
-  @staticmethod
-  def import_model_class(model_class):
-    class_name = re.sub(r'(?<!^)(?=[A-Z])', '_', model_class).lower()
-    return getattr(importlib.import_module('app.main.lib.shared_models.%s' % class_name), model_class)
-
-  @staticmethod
-  def start_server(model_class, model_key, options={}):
-    class_ = SharedModel.import_model_class(model_class)
-    instance = class_(model_key, options)
-    instance.load()
-    app.logger.info('[%s] Serving model...', model_key)
-    r = redis.Redis(host=app.config['REDIS_HOST'], port=app.config['REDIS_PORT'], db=app.config['REDIS_DATABASE'])
-    r.set('SharedModel:%s' % model_key, json.dumps({
-      'model_class': model_class,
-      'model_key': model_key,
-      'options': options
-    }))
-    r.sadd('SharedModel', model_key)
-    instance.bulk_run()
-
-  @staticmethod
-  def get_servers():
-    r = redis.Redis(host=app.config['REDIS_HOST'], port=app.config['REDIS_PORT'], db=app.config['REDIS_DATABASE'])
-    return [server.decode('utf-8') for server in r.smembers('SharedModel')]
-
-  @staticmethod
-  def get_client(model_key, options={}):
-    r = redis.Redis(host=app.config['REDIS_HOST'], port=app.config['REDIS_PORT'], db=app.config['REDIS_DATABASE'])
-    model_class = json.loads(r.get('SharedModel:%s' % model_key).decode('utf-8'))['model_class']
-    class_ = SharedModel.import_model_class(model_class)
-    return class_(model_key, options)
-
-  def __init__(self, model_key, options={}):
-    self.options = options
-    self.queue_name = model_key
-    self.datastore = redis.Redis(host=app.config['REDIS_HOST'], port=app.config['REDIS_PORT'], db=app.config['REDIS_DATABASE'])
-
-  def get_task(self, timeout=0):
-    item = self.datastore.blpop(self.queue_name, timeout)
-    if item:
-      task = Task(**json.loads(item[1].decode('utf-8')))
-      app.logger.info('[%s] Picking up task %s', self.queue_name, task.task_id)
-      return task
-
-  def send_response(self, task, response):
-    self.datastore.set(task.task_id, json.dumps({"response": response}))
-
-  def run(self):
-    while True:
-      task = self.get_task()
-      if task:
-        self.send_response(
-          task,
-          self.respond(task.task_package)
+    @staticmethod
+    def import_model_class(model_class):
+        class_name = re.sub(r"(?<!^)(?=[A-Z])", "_", model_class).lower()
```

Comment:
What would be the source of a model class name where it could even potentially fall afoul of this list? Is it free text entry at some point upstream of here?